skyrl-train/docs/examples/mini_swe_agent.rst

```diff
@@ -0,0 +1,109 @@
+    - Store the generated git patch after generation completes.
+2. Evaluation: 
+    - Initialize a fresh environment for the given instance using the given backend.
+    - Apply the model's git patch to the working directory in the environment.
+    - Run the evaluation script for the instance. If the script runs successfully, the instance is considered to be resolved, and unresolved otherwise.
+
+By running this workflow as a Ray task, we are also able to scale up generation across all the nodes in the Ray cluster. 
+
+
+ At a high level, the code looks as follows:
+
+.. code-block:: python
+
+    @ray.remote(num_cpus=0.01)
+    def init_and_run(instance: dict, litellm_model_name: str, sweagent_config: dict, data_source: str, ...):
+        model = get_model(litellm_model_name, sweagent_config.get("model", {}))
+        error = None
+        try:
+        env = get_sb_environment(sweagent_config, instance, data_source)
+        agent = DefaultAgent(model, env, **sweagent_config.get("agent", {}))
+        exit_status, model_patch = agent.run(instance["problem_statement"])
+        eval_result = evaluate_trajectory(instance, model_patch, sweagent_config, data_source)
+        except Exception as e:
+            error = str(e)
+        return agent.messages, eval_result, error
+
+    class MiniSweAgentGenerator(SkyRLGymGenerator):
+        async def generate_trajectory(self, prompt, env_extras, ...): 
+            messages, eval_result, error = init_and_run.remote(env_extras["instance"], ...)
+            ...
+
+Note that the full implementation has some additional logic for configuration and error handling, and can be found in :code_link:`examples/mini_swe_agent/mini_swe_generator.py`.
+
+
+Dataset preparation
+-------------------
+
+For training, we use `SWE-Gym <https://huggingface.co/SWE-Gym>`_, and more specifically the subset of SWE-Gym in `SumanthRH/SWE-Gym-Subset <https://huggingface.co/datasets/SumanthRH/SWE-Gym-Subset>`_.
+
+Execute the following command: 
+
+.. code-block:: bash
+
+    # execute from skyrl-train directory
+    uv run --isolated examples/mini_swe_agent/preprocess_swegym.py --output_dir ~/data/swe_gym_subset
+
+
+Training
+---------
+
+Prerequisites: Ensure that you have the required environment backend installed for generating trajectories with Mini-SWE-Agent. By default, we use `Podman <https://podman.io/docs>`_. This can be modified in :code_link:`examples/mini_swe_agent/swebench.yaml` 
+
+We provide two example scripts: One for Qwen3-8B model and another for the `Qwen/Qwen3-Coder-30B-A3B-Instruct <https://huggingface.co/Qwen/Qwen3-Coder-30B-A3B-Instruct>` model. While the first script for Qwen3-8B requires a single 8xH100 node, the script for the 30B model requires 2 8xH100 nodes for training.
+
+.. code-block:: bash
+
+    # execute from skyrl-train directory
+    bash examples/mini_swe_agent/run_mini_swe_8B.sh
+    # or for 32B:
+    # bash examples/mini_swe_agent/run_mini_swe_32B.sh
```

Comment:
Curious why this script is called 32B when it is 30B-A3B, if we use this convention elsewhere too that's ok, but I found it a little confusing (since there is also https://huggingface.co/Qwen/Qwen3-32B)